packages/twenty-front/src/pages/settings/developers/webhooks/components/SettingsDevelopersWebhookDetail.tsx

```diff
@@ -77,32 +92,50 @@ export const SettingsDevelopersWebhooksDetail = () => {
 
   const isAnalyticsV2Enabled = useIsFeatureEnabled('IS_ANALYTICS_V2_ENABLED');
 
-  const fieldTypeOptions = [
-    { value: '*', label: 'All Objects' },
-    ...objectMetadataItems.map((item) => ({
-      value: item.nameSingular,
-      label: item.labelSingular,
-    })),
+  const fieldTypeOptions: SelectOption<string>[] = useMemo(
+    () => [
+      { value: '*', label: 'All Objects', Icon: IconNorthStar },
+      ...objectMetadataItems.map((item) => ({
+        value: item.nameSingular,
+        label: item.labelSingular,
+        Icon: getIcon(item.icon),
+      })),
+    ],
+    [objectMetadataItems, getIcon],
+  );
+
+  const actionOptions: SelectOption<string>[] = [
+    { value: '*', label: 'All Actions', Icon: IconNorthStar },
+    { value: 'create', label: 'Create', Icon: IconPlus },
+    { value: 'update', label: 'Update', Icon: IconRefresh },
+    { value: 'delete', label: 'Delete', Icon: IconTrash },
   ];
 
   const { updateOneRecord } = useUpdateOneRecord<Webhook>({
     objectNameSingular: CoreObjectNameSingular.Webhook,
   });
 
+  const isValidOperation = (op: { object: string; action: string }) =>
+    op.object !== '' && op.action !== '';
+
+  const areAllOperationsValid = operations.every(isValidOperation);
+
   const handleSave = async () => {
-    setIsDirty(false);
-    await updateOneRecord({
-      idToUpdate: webhookId,
-      updateOneRecordInput: {
-        operations: operations.map((op) => `${op.object}.${op.action}`),
-        description: description,
-      },
-    });
-    navigate(developerPath);
+    if (areAllOperationsValid) {
+      setIsDirty(false);
+      await updateOneRecord({
+        idToUpdate: webhookId,
+        updateOneRecordInput: {
+          operations: operations.map((op) => `${op.object}.${op.action}`),
+          description: description,
+        },
+      });
+      navigate(developerPath);
+    }
```

Comment:
Instead of disabling the update, I would just clean the operations

`operations.filter((op) => isDefined(op.object) && isDefined(op.action)).map((op) => `${op.object}.${op.action}``

Maybe we can put that in a `cleanAndFormatOperations` function